### PR TITLE
plugins/telescope.file_browser: init

### PR DESCRIPTION
**Pull request by Wolbyte**
Added telescope extension: [telescope-file-browser](https://github.com/nvim-telescope/telescope-file-browser.nvim)
Fixes: #467

### CODE REVIEW

File: plugins/telescope/file-browser.nix [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    '';
+
+    respectGitignore = helpers.defaultNullOpts.mkBool false ''
+      induces slow-down w/ plenary finder (true if `fd` available)
+    '';
+
+    browseFiles = helpers.defaultNullOpts.mkNullable helpers.rawType "fb_finders.browse_files" ''
+      a custom lua function to override for the file browser
+    '';
+    browseFolders = helpers.defaultNullOpts.mkNullable helpers.rawType "fb_finders.browse_folders" ''
```
Comment (GaetanLepage) [inline plugins/telescope/file-browser.nix:81]:
```suggestion
    '';

    browseFolders = helpers.defaultNullOpts.mkNullable helpers.rawType "fb_finders.browse_folders" ''
```

File: plugins/telescope/file-browser.nix [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+      a custom lua function to override for the folder browser
+    '';
+
```
Comment (GaetanLepage) [inline plugins/telescope/file-browser.nix:84]: If some lua code is expected, we usually use a `str` option and we automatically perform the conversion to `rawType` in the implementation.

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Thank you for this addition !
Here are a few minor remarks.

File: plugins/telescope/file-browser.nix (hunk 2 above)
Comment (GaetanLepage) [reply plugins/telescope/file-browser.nix:84]: See [this example](https://github.com/pta2002/nixvim/blob/4a9a3c1f6cea52d756d406d6467b48898cdbff07/plugins/utils/nvim-ufo.nix#L63) for instance.

**Comment by Wolbyte** [comment]
Thank you for your review, I've applied the requested changes.

File: plugins/telescope/file-browser.nix [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+      );
+      hide_parent_dir = hideParentDir;
+      hijack_netrw = hijackNetrw;
+      mappings = helpers.ifNonNull' mappings (
+        with builtins;
+          mapAttrs (
+            _:
+              mapAttrs (
+                _: action:
+                  if typeOf action == "string"
```
Comment (GaetanLepage) [inline plugins/telescope/file-browser.nix:192]:
```suggestion
                  if isString action
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

File: plugins/telescope/file-browser.nix [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+in {
+  options.plugins.telescope.extensions.file_browser = {
+    enable = mkEnableOption "File browser extension for telescope";
+
+    package = helpers.mkPackageOption "telescope file_browser" pkgs.vimPlugins.telescope-file-browser-nvim;
+
+    theme = helpers.mkNullOrOption types.str "Custom theme, will use your global theme by default.";
+
+    path = helpers.defaultNullOpts.mkNullable (types.either types.str helpers.rawType) "vim.loop.cwd()" ''
+      dir to browse files from, `vim.fn.expanded` automatically
```
Comment (GaetanLepage) [inline plugins/telescope/file-browser.nix:35]:
```suggestion
      Directory to browse files from.
      `vim.fn.expanded` automatically.
```

Could you start comments with a capital letter and end them with a period please ?
Sorry for bothering you with that but it is to keep the documentation coherent.

Comment (Wolbyte) [reply plugins/telescope/file-browser.nix:35]: Sure thing, i tried to use other telescope extensions as a reference and it seems that wasn't a good choice lol

**Review by GaetanLepage — APPROVED** [review]
Thank you for the corrections !
LGTM

**Pull request merged**